tests/combine_samples.C

```diff
@@ -0,0 +1,200 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: This is a file function that reads snapshot or basis files
+//		and computes the SVD over the group. Also provides the option
+//		to subtract the mean of the group before computing the SVD.
+//
+// Assumptions: You are running this file with the same number of processors
+//		as when the snapshots/bases were saved.
+//
+// Inputs:
+//	[1] List of file names without extension.
+//	[2] Optional: "-b" or "basis" to compute SVD over bases instead 
+//	    of snapshots. (Snapshots are default.)
+// 	[3] Optional: "-m" or "mean" to subtract the mean before 
+//	    computing the SVD. (No subtraction is default.)
+//      [4] Optional: "-o" or "offset" to subtract an arbitrary offset vector
+//          before computing the SVD. Offset vector file supplied as next argument.
+//
+// Outputs:
+//	[1] total_snapshot.*:           single hdf5 file (per rank) of all loaded data.
+//	[2] mean.*:                     single hdf5 file (per rank) with subtracted mean.
+//	[3] total.* OR total_meansub.*: single hdf5 file (per rank) with the new POD basis.
+//
+// Examples: mpirun -n 2 ./combine_samples file1 file2 file3 -b -m
+//           mpirun -n 2 ./combine_samples -o offsetfile file1 file2 file3
+
+#include "linalg/BasisGenerator.h"
+#include "linalg/scalapack_wrapper.h"
+#include "linalg/BasisReader.h"
+#include "linalg/BasisWriter.h"
+#include <vector>
+#include "mpi.h"
+#include <string>
+#include <stdio.h>
+
+int main(int argc, char* argv[])
+{
+    // Initialize MPI
+    MPI_Init(&argc, &argv);
+    int size;
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+	
+    std::vector<std::string> sample_names;
+    int snaps = 0; 
+    int dim   = 0;
+    bool subtract_mean = false;
+    bool subtract_offset = false;
+    std::string kind = "snapshot";
```

Comment:
Not a big deal, but `kind` could be an `enum`.

Reply:
I see now that this string is used in `BasisGenerator`, so we can leave this as a string.